### ISSUE TITLE
Project loading without solution file

### ISSUE THREAD

**Issue by wjk** [OmniSharp/omnisharp-roslyn#634]
I am trying to write and maintain a fairly large codebase that uses C# `*.csproj` files. However, since I am doing this on macOS and therefore have no access to “full” Visual Studio, I cannot (easily) write a solution file. The problem is, without a solution file, OmniSharp won’t recognize the `*.csproj` files. Could OmniSharp be modified to support loading a `*.csproj` file by itself, one at a time, without needing a solution? (Of course, if a `*.csproj` file is loaded by itself, its project and assembly references would be recursively parsed to provide IntellliSense.)

**Comment by willl** [comment]
Maybe try using this to generate an .sln file? https://github.com/sriramgd/create-vs-sln-from-proj-files/

As for your other question, yes, it would be possible to modify Omnisharp to search for .csproj files. In fact I suspect we already have [code](https://github.com/OmniSharp/omnisharp-roslyn/blob/dev/src/OmniSharp.DotNet/Projects/ProjectSearcher.cs) to do so.

**Comment by wjk** [comment]
@willl That won’t work for me, as it doesn’t run on Mac.

**Comment by willl** [comment]
@wjk Perhaps you could use MonoDevelop/Xamarin Studio to generate a .sln file?

**Comment by wjk** [comment]
@willl I really would prefer to avoid using Xamarin Studio if it can be helped, but between this and continuous problems building Xamarin.Mac, I might just have to.

**Comment by willl** [comment]
Looks like there is a new PR underway to do this: https://github.com/OmniSharp/omnisharp-roslyn/pull/635

**Comment by DustinCampbell** [comment]
yup! #635 should address this

**Comment by DustinCampbell** [comment]
At least, #635 should address this for OmniSharp. @wjk: it's not clear what editor you're using with OmniSharp from your report. The editor would need to be updated to take advantage of this PR.

**Comment by wjk** [comment]
@DustinCampbell It’s VSCode. If I understand properly, all I need to do is set the `omnisharp.path` VSCode preference to point to an updated version of OmniSharp.

**Comment by DustinCampbell** [comment]
The C# Extension for VS Code will need to be taught to launch OmniSharp when there's just a folder of .csproj files. Today, it doesn't do that. I'll be updating [this code](https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/omnisharp/launcher.ts#L52-L128) in the next few days after #635 is merged in to take advantage of the changes.

**Comment by DustinCampbell** [comment]
I just submitted https://github.com/OmniSharp/omnisharp-vscode/pull/760 for the VS Code extension to better process launch targets for .csproj files when no .sln is present.

**Comment by DustinCampbell** [comment]
Re-opening this issue until the change in the omnisharp-vscode repo is merged.

**Comment by DustinCampbell** [comment]
[v1.5-beta1](https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.5-beta1) of the C# extension has been released which can launch OmniSharp on a folder of .csproj files when no .sln file is present.
